src/backend/distributed/commands/schema.c

```diff
@@ -139,3 +145,69 @@ PreprocessAlterTableSchemaStmt(Node *node, const char *queryString)
 
 	return NIL;
 }
+
+
+/*
+ * PreprocessGrantOnSchemaStmt is executed before the statement is applied to the local
+ * postgres instance.
+ *
+ * In this stage we can prepare the commands that need to be run on all workers to grant
+ * on schemas. Only grant statements for distributed schema are propagated.
+ */
+List *
+PreprocessGrantOnSchemaStmt(Node *node, const char *queryString)
+{
+	GrantStmt *stmt = castNode(GrantStmt, node);
+	Assert(stmt->objtype == OBJECT_SCHEMA);
+
+	List * distributedSchemas = FilterDistributedSchemas(stmt->objects);
+
+	if (distributedSchemas == NIL)
```

Comment:
`list_length(...) == 0` is a better check here. A list can be empty without being `NIL`